state/logs.go

```diff
@@ -854,6 +855,29 @@ func initLogsSession(st ModelSessioner) (*mgo.Session, *mgo.Collection) {
 	return session, db.C(logCollectionName(st.ModelUUID()))
 }
 
+// dbCollectionSizeToInt processes the result of Database.collStats()
+func dbCollectionSizeToInt(result bson.M, collectionName string) (int, error) {
+	size, ok := result["size"]
+	if !ok {
+		logger.Warningf("mongo collStats did not return a size field for %q", collectionName)
+		// this wasn't considered an error in the past, just treat it as size 0
+		return 0, nil
+	}
+	if asint, ok := size.(int); ok {
+		return asint, nil
+	}
+	if asint64, ok := size.(int64); ok {
+		// 2billion megabytes is 2 petabytes, which is outside our range anyway.
+		if asint64 > math.MaxInt32 {
+			return math.MaxInt32, nil
+		}
```

Comment:
Is there ever a case that `asint64 < 0` so that we also handle the min case or `math.MinInt32` case?

Reply:
Well, if mongo returns a negative number of megabytes for the size of a collection we certainly have issues. I'm ok handling that case, just in a belt-and-suspenders type of defensive programming. But I'd probably cap the threshold at 0. Cause a negative size would do *really* weird things.